### PR TITLE
ZCS-3456:Attendee's status isn't updated

### PR DESCRIPTION
**Pull request by ykomiyam**
[Bug]
Organizer see the attendee's status as Tentative instead of Accepted
even when the attendee proposed a new time in invite and the organizer
accepted it.

[Root cause]
The the CounterAppointmentRequest proposed by the attendee contains
the attendee info ("PROPERTY:ATTENDEE(ATTENDEE)"), but it does not
contain the data of RSVP in it; that should be acceptable, and expected
behaviour because such an information has not been updated on the attendee
side.  On the other hand, when the organizer received the counter
appointment (via email), if there is no RSVP data in the attendee info,
the organizer turned the RSVP off for to the corresponding appointment
object.  Therefore, if the attendee later sent the acceptance, the
organizer ignored it even the acceptance email asked to update the
RSVP data on the organizer ("updateOrganizer":"TRUE").

[Fix]
Changed the default value of the RSVP data to true.

### CODE REVIEW

File: store/src/java-test/com/zimbra/cs/mailbox/calendar/Calendar_NewMeetingRequest_RsvpTrue.txt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+DTSTART:16010101T000000
+TZOFFSETTO:+0900
+TZOFFSETFROM:+0900
+END:STANDARD
+END:VTIMEZONE
+BEGIN:VEVENT
+UID:8282cdfc-a5a6-4ed4-84e2-be12c2bc4af8
+SUMMARY:Party
+ATTENDEE;CN=attendee1;ROLE=REQ-PARTICIPANT;PARTSTAT=NEEDS-ACTION;RSVP=TRUE:mailto
+ :attendee1@zimbra.com
```
Comment (jiteshsojitra) [inline store/src/java-test/com/zimbra/cs/mailbox/calendar/Calendar_NewMeetingRequest_RsvpTrue.txt]: I would suggest not to use @zimbra.com for test accounts for public code.

File: store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java [hunk 2]
```diff
@@ -16,16 +16,44 @@
...
+import com.zimbra.cs.mailbox.MailboxTestUtil;
+import com.zimbra.cs.mime.ParsedMessage;
+import com.zimbra.cs.mime.ParsedMessage.CalendarPartInfo;
 
 public class ZAttendeeTest {
 
+    @BeforeClass
+    public static void init() throws Exception {
+        MailboxTestUtil.initServer();
+        Provisioning prov = Provisioning.getInstance();
```
Comment (rupalid) [inline store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java]: Move the account creation to setUp

File: store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java [hunk 3]
```diff
@@ -16,16 +16,44 @@
...
 import com.zimbra.common.calendar.ZCalendar.ZParameter;
 import com.zimbra.common.calendar.ZCalendar.ZProperty;
 import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.util.ByteUtil;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.mailbox.MailboxTestUtil;
+import com.zimbra.cs.mime.ParsedMessage;
+import com.zimbra.cs.mime.ParsedMessage.CalendarPartInfo;
 
 public class ZAttendeeTest {
```
Comment (rupalid) [inline store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java]: Can you add the following, so we can find which test fails in Circle CI
 @Rule public TestName testName = new TestName();
    @Rule public MethodRule watchman = new ZTestWatchman();

File: store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java [hunk 4]
```diff
@@ -16,16 +16,44 @@
...
+    public static void init() throws Exception {
+        MailboxTestUtil.initServer();
+        Provisioning prov = Provisioning.getInstance();
+
+        prov.createAccount("test@zimbra.com", "secret", Maps.<String, Object>newHashMap());
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        MailboxTestUtil.clearData();
```
Comment (rupalid) [inline store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java]: Please also add the following , so we can know which test fails in CircleCI

System.out.println(testName.getMethodName());

File: store/src/java-test/com/zimbra/cs/mailbox/calendar/Calendar_NewMeetingRequest_RsvpTrue.txt (hunk 1 above)
Comment (ykomiyam) [reply store/src/java-test/com/zimbra/cs/mailbox/calendar/Calendar_NewMeetingRequest_RsvpTrue.txt]: Fixed.  Use "@testdomain.com"

File: store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java (hunk 3 above)
Comment (ykomiyam) [reply store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java]: Fixed.

File: store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java (hunk 2 above)
Comment (ykomiyam) [reply store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java]: Fixed.

File: store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java (hunk 4 above)
Comment (ykomiyam) [reply store/src/java-test/com/zimbra/cs/mailbox/calendar/ZAttendeeTest.java]: Fixed.

File: store/src/java/com/zimbra/cs/mailbox/calendar/ZAttendee.java [hunk 5]
```diff
@@ -142,10 +142,10 @@ public ZAttendee(ZProperty prop) {
         setRole(prop.paramVal(ICalTok.ROLE, null));
         setPartStat(prop.paramVal(ICalTok.PARTSTAT, null));
 
-        String rsvpStr = prop.paramVal(ICalTok.RSVP, "FALSE");
-        boolean rsvp = false;
-        if (rsvpStr.equalsIgnoreCase("TRUE"))
-            rsvp = true;
+        String rsvpStr = prop.paramVal(ICalTok.RSVP, "TRUE");
+        boolean rsvp = true;
```
Comment (chinmay15) [inline store/src/java/com/zimbra/cs/mailbox/calendar/ZAttendee.java]: this is setting rsvp to true by default. did you test by sending appointment with rsvp=false?

File: store/src/java/com/zimbra/cs/mailbox/calendar/ZAttendee.java [hunk 6]
```diff
@@ -142,10 +142,10 @@ public ZAttendee(ZProperty prop) {
         setRole(prop.paramVal(ICalTok.ROLE, null));
         setPartStat(prop.paramVal(ICalTok.PARTSTAT, null));
 
-        String rsvpStr = prop.paramVal(ICalTok.RSVP, "FALSE");
-        boolean rsvp = false;
-        if (rsvpStr.equalsIgnoreCase("TRUE"))
-            rsvp = true;
+        String rsvpStr = prop.paramVal(ICalTok.RSVP, "TRUE");
```
Comment (fatkudu) [inline store/src/java/com/zimbra/cs/mailbox/calendar/ZAttendee.java]: See https://tools.ietf.org/html/rfc2445#section-4.2.17 - the default value of RSVP is FALSE.
I think this change is too intrusive and may lead to bugs.  Presumably, the attendee information from the REPLY is at some point being applied to the ORGANIZER's calendar entry.  The code that does that change needs to consider what should be taken from the reply and what should not.

**Comment by ykomiyam** [comment]
Retract the pull request as per Gren's comment

**Pull request closed without merging**